src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/memories/JqlMemory2.kt

```diff
@@ -0,0 +1,12 @@
+package com.atlassian.performance.tools.jiraactions.api.memories
+
+import com.atlassian.performance.tools.jiraactions.api.jql.Jql
+import com.atlassian.performance.tools.jiraactions.api.page.IssuePage
+import java.util.function.Predicate
+
+interface JqlMemory2 : Memory<Jql> {
```

Comment:
If this a replacement for `JqlMemory`, shouldn't that one be deprecated?
If it's too early to deprecate, then maybe it's too early to put in replacements? New API works best if it was tried in some way, e.g.:
* unit/integration tests within the module
* equivalent API used outside of the module